### PR TITLE
Make changes to bump version

### PR DESCRIPTION
**Pull request by jlapeyre**
Bump version to 0.43.11

### CODE REVIEW

**Review by Eric-Arellano — DISMISSED** [review]

**Comment by jlapeyre** [comment]
No, the reno report includes a heading that must be omitted when pasting in  ~~Looks like some doc text may have been pasted twice. Checking ...~~

**Review by Eric-Arellano — APPROVED** [review]

**Comment by jlapeyre** [comment]
I clicked the "squash and merge" button. But there was no opportunity (that I saw) to squash. So the commit message of the merge commit is not descriptive of the tagged version.
I'm reluctant to tag this commit with the new version and release.
Probably better to make a clean commit history on monday and tag the last commit.

**Comment by Eric-Arellano** [comment]
> I clicked the "squash and merge" button. But there was no opportunity (that I saw) to squash. So the commit message of the merge commit is not descriptive of the tagged version.

It squash merged, but we need to fix our config for this repo so that squash merge defaults to the PR Title And Description. I'm not an admin so haven't been able to.

> I'm reluctant to tag this commit with the new version and release.
Probably better to make a clean commit history on monday and tag the last commit.

No, I think you're fine. The commit message itself isn't that crucial because we have the Git tag and GitHub releases to make it clear that that single squashed commit is where the release happened. I doubt many people are looking at the commit message, and instead they care about the tag and GitHub Release.

**Comment by jlapeyre** [comment]
Ok. I'll just tag what we have. (I responded to you in two minutes, and late last night within 5 minutes. That is fortuitous. I'm not staring at this PR day and night)

File: docs/release_notes.rst [hunk 1]
```diff
@@ -21,6 +21,139 @@ This table tracks the meta-package versions and the version of each Qiskit eleme
...
 ###############
 
+*************
+Qiskit 0.43.1
+*************
+
+.. _Release Notes_Terra_0.24.1:
+
+0.24.1
+======
```
Comment (jakelishman) [inline docs/release_notes.rst:31]: This isn't quite right - we ought to have modified the title to say "Terra 0.24.1", and to have included
```rst
Aer 0.12.0
==========

No change.


IBMQ Provider 0.x.y
===================

No change.
```
below.  We can just fix this up though, even if the metapackage was already deployed - it'll just get folded into the documentation.

Comment (jlapeyre) [reply docs/release_notes.rst:31]: Yes, you are correct. Can this be fixed up with this release? I don't see how. Maybe retroactively fix it along with the next bugfix release.

Comment (jakelishman) [reply docs/release_notes.rst:31]: We can make changes to the docs whenever, and they'll get deployed as-and-when.  The main `qiskit.org/documentation` page should get updated immediately with new files from this repo, unless the process has changed.

Comment (Eric-Arellano) [reply docs/release_notes.rst:31]: That is still correct that we deploy docs on every merge.

Comment (jakelishman) [reply docs/release_notes.rst:31]: Thanks Eric.  John, would you be able to make a new PR fixing things up?

Comment (jlapeyre) [reply docs/release_notes.rst:31]: Yes. I'll make a new PR.

**Pull request merged**